src/coreclr/jit/importer.cpp

```diff
@@ -19631,6 +19631,21 @@ void Compiler::impInlineInitVars(InlineInfo* pInlineInfo)
         }
 
         GenTree* actualArg = use.GetNode();
+        if (actualArg != nullptr && actualArg->OperIsSimple())
+        {
+            GenTree* op1 = actualArg->gtGetOp1();
+            GenTree* op2 = actualArg->gtGetOp2IfPresent();
+            if (actualArg->OperIsUnary() && (op1 != nullptr) && op1->OperIsConst())
+            {
+                actualArg = gtFoldExprConst(actualArg);
+            }
+            else if (actualArg->OperIsBinary() && (op1 != nullptr) && (op2 != nullptr) && op1->OperIsConst() &&
+                     op2->OperIsConst())
+            {
+                actualArg = gtFoldExprConst(actualArg);
+            }
+        }
+
```

Comment:
Any reason we can't just call `gtFoldExpr(actualArg)` and catch some of the other cases where we can fold?

If you still want to case it out like this, I don't think any of these `!= nullptr` checks are needed.

Reply:
the was some assert that checked actualArg's type against sigType. E.g. it failed for int -> ubyte (after `gtFoldExpr`)
```
    *  ADD       int
    +--*  CNS_INT   int    0
    \--*  FIELD     ubyte  hackishFieldName
        \--*  LCL_VAR   ref    V00 this
```
after `gtFoldExpr`:
```
    *  FIELD     ubyte  hackishFieldName
    \--*  LCL_VAR   ref    V00 this
```
I removed the assert and it seems work fine.

> If you still want to case it out like this, I don't think any of these != nullptr checks are needed.

E.g. GT_HWINTRINSIC (Vector_get_Zero) is a binary node with both children nullptr (IsBinary returns true)